kong/dao/schemas/upstreams.lua

```diff
@@ -1,10 +1,124 @@
 local Errors = require "kong.dao.errors"
 local utils = require "kong.tools.utils"
+local match = string.match
+local sub = string.sub
 
 local DEFAULT_SLOTS = 100
 local SLOTS_MIN, SLOTS_MAX = 10, 2^16
 local SLOTS_MSG = "number of slots must be between " .. SLOTS_MIN .. " and " .. SLOTS_MAX
 
+
+local function check_nonnegative(arg)
+  if arg < 0 then
+    return false, "must be greater than or equal to 0"
+  end
+end
+
+
+local function check_positive_int(t)
+  if t < 1 or t > 2^31 - 1 or math.floor(t) ~= t then
+    return false, "must be an integer between 1 and " .. 2^31 - 1
+  end
+
+  return true
+end
+
+
+local function check_http_path(arg)
+  if match(arg, "^%s*$") then
+    return false, "path is empty"
+  end
+  if sub(arg, 1, 1) ~= "/" then
+    return false, "must be prefixed with slash"
+  end
+  return true
+end
+
+
+local function check_http_statuses(arg)
```

Comment:
could we also apply `check_positive_int` in this check as well?

Reply:
Why? The only logic I can see the `check_positive_int()` validator enforcing would be checking if the argument is an integer. Is that what you are suggesting we should check in this validator? Otherwise, the `check_http_statuses()` validator already has narrower acceptance rules regarding the allowed range.

Reply:
Not saying this isn't an invalid check to perform, ofc, just making sure.